src/App.tsx

```diff
@@ -1,17 +1,62 @@
 /* eslint-disable jsx-a11y/control-has-associated-label */
-import React, { useContext, useEffect, useRef } from 'react';
+import React, {
+  useContext,
+  useEffect,
+  useRef,
+  useState,
+} from 'react';
+import { getTodos } from './api/todos';
 import { AuthContext } from './components/Auth/AuthContext';
+import { ErrorNotification } from './components/ErrorNotification';
+import { Footer } from './components/Footer';
+import { TodoList } from './components/TodoList';
+import { FilterStatus } from './types/FilterStatus';
+import { Todo } from './types/Todo';
 
 export const App: React.FC = () => {
   // eslint-disable-next-line @typescript-eslint/no-unused-vars
   const user = useContext(AuthContext);
   const newTodoField = useRef<HTMLInputElement>(null);
+  const [todos, setTodos] = useState<Todo[]>([]);
+  const [error, setError] = useState(false);
+  const [filterStatus, setFilterStatus] = useState('all');
+
+  const filteredTodos = todos.filter(todo => {
+    switch (filterStatus) {
+      case FilterStatus.All:
+        return todo;
+
+      case FilterStatus.Active:
+        return todo.completed === false;
+
+      case FilterStatus.Completed:
+        return todo.completed;
+
+      default:
+        return null;
+    }
+  });
+  let userId = 0;
+
+  if (user?.id) {
+    userId = user.id;
+  }
```

Comment:
better check `user?.id` inside `useEffect` and fetch todos if the user exists and just return if not (your app can't work without user)